api4/team.go

```diff
@@ -1275,12 +1275,23 @@ func inviteUsersToTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	emailList := model.ArrayFromJSON(r.Body)
+	memberInvite := &model.MemberInvite{}
+	if jsonErr := json.NewDecoder(r.Body).Decode(memberInvite); jsonErr != nil {
+		c.Err = model.NewAppError("Api4.inviteUsersToTeams", "api.team.invite_members_to_team_and_channels.invalid_body.app_error", nil, jsonErr.Error(), http.StatusBadRequest)
+		return
+	}
+
+	emailList := memberInvite.Emails
 
 	for i := range emailList {
 		emailList[i] = strings.ToLower(emailList[i])
 	}
 
+	if err := memberInvite.IsValid(); err != nil {
+		c.Err = err
+		return
+	}
```

Comment:
The `isValid` seems to be checking if all emails are valid and failing if any are. I believe this would break the send gracefuly functionality given that it will not send any if there is one that is not valid. Before it would send all that it could and return the error of which ones failed.

Reply:
good catch! thanks Maria, with this validation the "gracefully" behavior is broken.